components/channel_groups_manage_modal/channel_groups_manage_modal.jsx

```diff
@@ -78,18 +107,32 @@ export default class ChannelGroupsManageModal extends React.PureComponent {
                         /></span>
                     </div>
                 </div>
-                <div className='more-modal__actions'>
-                    <button
-                        id='removeMember'
-                        type='button'
-                        className='btn btn-danger btn-message'
-                        onClick={() => this.onClickRemoveGroup(item, listModal)}
-                    >
-                        <FormattedMessage
-                            id='group_list_modal.removeGroupButton'
-                            defaultMessage='Remove Group'
-                        />
-                    </button>
+                <div className='more-modal__dropdown'>
+                    <MenuWrapper>
+                        <a>
+                            <span>{title} </span>
+                            <span className='caret'/>
+                        </a>
+                        <Menu
+                            openLeft={true}
+                            ariaLabel={Utils.localizeMessage('channel_members_dropdown.menuAriaLabel', 'Channel member role change')}
+                        >
+                            <Menu.ItemAction
+                                show={!item.scheme_admin}
+                                onClick={() => this.makeChanneldmin(item, listModal)}
```

Comment:
Defining the `onClick` function inline like this in the JSX re-defines the function upon each re-render which in turn causes a re-render of the menu item. An alternative is to define the function(s) outside of the JSX.

Reply:
Yes, the arrow functions get recreated each render, however, from my debugging/tracing, the Menu render() gets invoked only once, when selecting the dropdown caret. It could be that what you are advocating is smth. similar to TeamList/TeamRow approach, however, to me that is beyond the current workitem scope. Given that the current arrow function approach is the pre-existing one in these files and looks to be quite common, that the menu has a limited number of menu-items (3) and that it doesn't look like chained rendering actually happens, I'm in favor keeping the current approach. Thoughts?

Reply:
Perhaps a good community item.